src/SourceGenerators/Uno.UI.Tasks/BatchMergeXaml/MergedDictionary.cs

```diff
@@ -0,0 +1,372 @@
+                // or create new ones if we've found new keys.
+                foreach (XmlNode childNode in node.ChildNodes)
+                {
+                    string nodeKey = GetKey(childNode);
+
+                    if (nodeKey == null || nodeKey.Length == 0)
+                    {
+                        continue;
+                    }
+                    else if (nodeKey == "Dark")
+                    {
+                        // If we don't specify the dictionary "Dark", then "Default" will be used instead.
+                        // Having both of those, however, will result in "Default" never being used, even
+                        // if it contains something that "Dark" does not.
+                        // Since we're merging everything into a single dictionary, we need to standardize
+                        // to only one of the two. Since most dictionaries use "Default", we'll go with that.
+                        nodeKey = "Default";
+                    }
+
+                    MergedDictionary mergedThemeDictionary = null;
+
+                    if (mergedThemeDictionaryByKeyDictionary.TryGetValue(nodeKey, out mergedThemeDictionary) == false)
+                    {
+                        mergedThemeDictionary = new MergedDictionary(owningDocument, this);
+                        mergedThemeDictionaryByKeyDictionary.Add(nodeKey, mergedThemeDictionary);
+                    }
+
+                    foreach (XmlNode resourceDictionaryChild in childNode.ChildNodes)
+                    {
+                        mergedThemeDictionary.AddNode(resourceDictionaryChild, xmlnsReplacementDictionary);
+                    }
+                }
+            }
+            else
+            {
+                // First, we need to check if this is a node with a key.  If it is, then we'll replace
+                // the previous node we saw with this key, in order to have only one entry per key.
+                // if it's not, or if we haven't seen a previous node with this key, then we'll just
+                // add it to our list.
+                string nodeKey = GetKey(node);
+
+                if (nodeKey.Length == 0 || !nodeKeyToNodeListIndexDictionary.ContainsKey(nodeKey))
+                {
+                    if (nodeKey.Length != 0)
+                    {
+                        nodeKeyToNodeListIndexDictionary.Add(nodeKey, nodeList.Count);
+                    }
+
+                    nodeList.Add(node);
+                }
+                else
+                {
+                    int previousNodeIndex = nodeKeyToNodeListIndexDictionary[nodeKey];
+                    nodeList[previousNodeIndex] = node;
+                }
+
+                if (nodeKey.Length > 0 && parentDictionary != null)
+                {
+                    parentDictionary.RemoveAncestorNodesWithKey(nodeKey);
+                }
+
+                if (nodeKey.Length > 0 && parentDictionary != null)
```

Comment:
Codacy found an issue: [This condition was just checked on line 194.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5930374)